HEInventions.Logging/RavenLog.cs

```diff
@@ -0,0 +1,219 @@
+        }
+
+        /// <summary>
+        /// Sentry event with severity level Info.
+        /// </summary>
+        /// <param name="msg">The error message.</param>
+        /// <param name="exception">The associated exception object.</param>
+        public void Info(String msg, Exception exception = null)
+        {
+            NewSentryEvent(msg, ErrorLevel.Info, exception);
+        }
+
+        /// <summary>
+        /// Sentry event with severity level Debug.
+        /// </summary>
+        /// <param name="msg">The error message.</param>
+        /// <param name="exception">The associated exception object.</param>
+        public void Debug(String msg, Exception exception = null)
+        {
+            NewSentryEvent(msg, ErrorLevel.Debug, exception);
+        }
+
+        /// <summary>
+        /// Sentry event with severity level Fatal.
+        /// </summary>
+        /// <param name="msg">The error message.</param>
+        /// <param name="exception">The associated exception object.</param>
+        public void Fatal(String msg, Exception exception = null)
+        {
+            NewSentryEvent(msg, ErrorLevel.Fatal, exception);
+        }
+        #endregion
+
+        #region Sentry event configuration
+        /// <summary>
+        /// Create a new Sentry event that includes a message, severity, and optional Exception data.
+        /// SharpRaven's ErrorLevel enum goes from Fatal = 0 to Debug = 4.
+        /// </summary>
+        /// <param name="msg">The message that will appear on the Sentry log entries.</param>
+        /// <param name="level">The error level attached to the Sentry event.</param>
+        /// <param name="ex">The (optional) Exception object which is added to the Sentry event.</param>
+        private void NewSentryEvent(String msg, ErrorLevel level, Exception ex)
+        {
+            if (ravenClient == null)
+                return;
+            try
+            {
+                var threshold = Enum.Parse(typeof(ErrorLevel), ErrorThreshold);
+                if ((int)level > (int)threshold)
+                    return;
+            }
+            catch (Exception e)
+            {
+                return;
+            }
+
+            SentryEvent sentryEvent;
+
+            if (ex != null)
+                sentryEvent = new SentryEvent(ex);
+            else
+                sentryEvent = new SentryEvent(msg);
+
+            sentryEvent.Tags = new Dictionary<string, string> { { "host_id", InstallationHostID } };
+            sentryEvent.Message = msg;
+            sentryEvent.Level = level;
+            ravenClient.Capture(sentryEvent);
+
+            OnMessage?.Invoke(sentryEvent);
+        }
+        #endregion
+
+        #region Parsing and utility methods
+        /// <summary>
+        /// Forms a hostname from the machine settings file (in JSON format) data.
+        /// </summary>
+        /// <param name="settingsFile">The machine settings JSON file.</param>
```

Comment:
Is this the PATH to the settings JSON file?

Reply:
updated comment